utils/src/ast-grep/package-json.ts

```diff
@@ -0,0 +1,78 @@
+import type { SgRoot, Edit } from "@codemod.com/jssg-types/main";
+
+/**
+ * Get the "scripts" node from a package.json AST.
+ * @param packageJsonRootNode The root node of the package.json AST.
+ * @returns The "scripts" node, or null if not found.
+ */
+export const getScriptsNode = (packageJsonRootNode: SgRoot) => {
+	const scriptsNodes = packageJsonRootNode
+		.root()
+		.findAll({
+			rule: {
+				kind: "pair",
+				has: {
+					field: "key",
+					kind: "string",
+					has: {
+						kind: "string_content",
+						regex: "scripts"
+					}
+				}
+			}
+		});
+
+	if (scriptsNodes.length > 1)
+		throw new Error(`Multiple "scripts" fields found in ${packageJsonRootNode.filename()}`);
+
+	return scriptsNodes[0] ?? null;
+};
```

Comment:
so the actual implementation returns

```
"scripts": {
  one: "example",
  two: "example",
}
```

that makes be necessary do `result.children()[2]`  to get the node with scripts itself, as the first is `scripts` json key and the second one is the column.

If we updated to approach below, it will return a list of nodes with the scripts something like that

```js
[
  one: "example", // as SgNode Object
  two: "example", // as SgnNode Object
]
```

This way, I think it will be easier to manipulate, and throwing an error when not found won’t be necessary, because if there are no scripts, it will just return an empty array.

```suggestion
export const getScriptsNode = (packageJsonRootNode: SgRoot) =>
	packageJsonRootNode.root().findAll({
		rule: {
			kind: "pair",
			inside: {
				kind: "object",
				inside: {
					kind: "pair",
					has: {
						field: "key",
						kind: "string",
						has: {
							kind: "string_content",
							regex: "scripts",
						},
					},
				},
			},
		},
	});
```

Reply:
I thought a little bit more about that, and maybe the key `scripts` can be received as a paramter and this function can be something like `getObjectFields` and can be used like


```js
getObjectFields(node, 'scripts')
getObjectFields(node, 'dependencies')
getObjectFields(node, 'workspaces')
getObjectFields(node, 'exports')
```

I do not have sure if getObjectFields is the best name, btw

Reply:
for now, we will just cover args thing